### PR TITLE
Fixed issues #136 and #375

### PR DESCRIPTION
**Pull request by ma-garcia**

### CODE REVIEW

File: src/somef/cli.py [hunk 1]
```diff
@@ -1655,8 +1655,23 @@ def merge(header_predictions, predictions, citations, citation_file_text, dois,
     for i in range(len(citations)):
         if 'citation' not in predictions.keys():
             predictions['citation'] = []
-        predictions['citation'].insert(0, {'excerpt': citations[i], 'confidence': [1.0],
-                                           'technique': 'Regular expression'})
+        if citations[i].find('https://doi.org/') >= 0 or citations[i].find('doi ') >= 0:
+            doi_text = ""
+            text_citation = citations[i]
+            if text_citation.find("https://doi.org/") >= 0:
```
Comment (dgarijo) [inline src/somef/cli.py:1661]: This is not entirely correct. Please add support for http://dx.doi.org.  Maybe the regular expression should be doi.org/ so we cover both, and also http and https

**Comment by dgarijo** [comment]
I left a small comment. Please have a look.

File: src/somef/cli.py (hunk 1 above)
Comment (dgarijo) [reply src/somef/cli.py:1661]: Also, in the citations there is bibtex sometimes. If that's the case, the doi may appear as:
`doi = {10.1142/S0218194022500036},`
Example: 
```
@article{osorio_dockerpedia_2022,
	title = {{DockerPedia}: {A} {Knowledge} {Graph} of {Software} {Images} and {Their} {Metadata}},
	volume = {32},
	issn = {0218-1940, 1793-6403},
	shorttitle = {{DockerPedia}},
	url = {https://dgarijo.com/papers/reproducibility_docker.pdf},
	doi = {10.1142/S0218194022500036},
	language = {en},
	number = {01},
	urldate = {2022-03-02},
	journal = {International Journal of Software Engineering and Knowledge Engineering},
	author = {Osorio, Maximiliano and Buil-Aranda, Carlos and Santana-Perez, Idafen and Garijo, Daniel},
	month = jan,
	year = {2022},
	pages = {71--89},
}
```

Comment (ma-garcia) [reply src/somef/cli.py:1661]: This is already done in the lines below:

>             elif text_citation.find("doi") >= 0:
>                 doi_pos = text_citation.find("doi")
>                 starts = text_citation[doi_pos:].find("{")
>                 ends = text_citation[starts + doi_pos:].find("}")

**Comment by ma-garcia** [comment]
Updated code to support http(s)://(dx.)doi.org pattern

**Pull request merged**